### PR TITLE
:fix rooms pageNum and return format

### LINKED ISSUE #144: GET /rooms response change `offset` to `perPage`
現在錯誤的回傳格式
```
Response 200
{
/* ... */
   "meta": {
    "page: int",
    "offset": int,    // <== expecting `perPage`, not `offset`
     "total": int,
   },
}
```

正確的回傳格式
```
Response 200
{
/* ... */
   "meta": {
    "page: int",
    "perPage": int,
     "total": int,
   },
}
```

### PR DESCRIPTION
**Pull request by m1a2st**
## Why need this change? / Root cause: 
- according to frontend's request
## Changes made:
- page logic and return api format
## Test Scope / Change impact:
- RoomControllerTest
## Issue
- resolve #144

### CODE REVIEW

File: spring/src/main/kotlin/tw/waterballsa/gaas/spring/controllers/RoomController.kt [hunk 1]
```diff
@@ -208,10 +208,11 @@ class RoomController(
         @field:Positive(message = "The perPage must be a positive number.")
         val perPage: Int
     ) {
+        // 這邊為了配合前端，因為他們的 page 是從 1 開始，所以這邊要減 1
```
Comment (Wally5077) [inline spring/src/main/kotlin/tw/waterballsa/gaas/spring/controllers/RoomController.kt:211]: 請前端他們帶上來 0

File: spring/src/main/kotlin/tw/waterballsa/gaas/spring/controllers/presenter/GetRoomsPresenter.kt [hunk 2]
```diff
@@ -42,7 +41,8 @@ private fun Room.Player.toGetRoomsView(): GetRoomsViewModel.RoomViewModel.Player
 
 private fun Pagination<Room>.toPage(size: Int): GetRoomsViewModel.Page =
     GetRoomsViewModel.Page(
-        page = page,
-        offset = offset,
+        // 這邊為了配合前端，因為他們的 page 是從 1 開始，所以這邊要加 1
```
Comment (Wally5077) [inline spring/src/main/kotlin/tw/waterballsa/gaas/spring/controllers/presenter/GetRoomsPresenter.kt:44]: ditto 請前端他們帶上來 0

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt [hunk 3]
```diff
@@ -139,10 +139,11 @@ class RoomControllerTest @Autowired constructor(
             "3", "test3@mail.com",
             "winner1234", "google-oauth2|200000000000000000000"
         )
-        val request = TestGetRoomsRequest("WAITING", 0, 10)
+        val request = TestGetRoomsRequest("WAITING", 1, 10)
 
         givenWaitingRooms(userB, userC)
         request.whenUserAVisitLobby(userA)
+            .andDo { println(it.response.contentAsString) }
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:146]: Ｑ：這是測試程式碼？

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/models/TestGetRoomsRequest.kt [hunk 4]
```diff
@@ -9,5 +9,6 @@ class TestGetRoomsRequest(
     val perPage: Int
 ) {
     fun toStatus() : Room.Status = Room.Status.valueOf(status)
-    fun toPagination(): Pagination<Any> = Pagination(page, perPage)
+    // 這邊為了配合前端，因為他們的 page 是從 1 開始，所以這邊要減 1
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/models/TestGetRoomsRequest.kt:12]: ditto 請前端他們帶上來 0

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt (hunk 3 above)
Comment (frankvicky) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:146]: Q：印出來的目的？

**Review by frankvicky — CHANGES_REQUESTED** [review]

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

**Review by Wally5077 — APPROVED** [review]
**LGTM**

**Review by frankvicky — APPROVED** [review]

**Pull request merged**